File: ShadowFrameworkLite20/src/shadow/image/bitmaps/SFBitmapFunctionPerlinNoise.java
package shadow.image.bitmaps;

public class SFBitmapFunctionPerlinNoise extends SFPerlinNoise{

	private SFBitmapFunction bitmap;
	
	public SFBitmapFunctionPerlinNoise(){
	}
	
	public SFBitmapFunctionPerlinNoise(int width,int height,float[] weights,boolean rgb){
		
		super();
		
		setWidth((short)width);
		setHeight((short)height);
		
		this.weights=weights;
		this.rgb=rgb;
	}
	
	
	public SFBitmapFunction getBitmap() {
		return bitmap;
	}

	public void setBitmap(SFBitmapFunction bitmap) {
		this.bitmap = bitmap;
	}

	
	@Override
	protected float getOctaveValue(float u, float v, int octave) {
		
		float U=u*octave;
		U-=((int)U);
		float V=v*octave;
		V-=((int)V);
		
		float value=bitmap.getValue(U, V);
		return value*256;
	}
	

}

File: trunk/ShadowFrameworkJS/src/shadow/image/bitmaps/SFBitmapFunctionPerlinNoise.js


function SFBitmapFunctionPerlinNoise(width,height,weights,rgb){
	this.bitmap=null;
	if(width!=undefined && height!=undefined && weights!=undefined && rgb!=undefined){
		
		this.setWidth(width);
		this.setHeight(height);
		
		this.weights=weights;
		this.rgb=rgb;
	}
	
}

inherit(SFBitmapFunctionPerlinNoise,SFPerlinNoise);


SFBitmapFunctionPerlinNoise.prototype["getBitmap"]=function(){
		return this.bitmap;
};

SFBitmapFunctionPerlinNoise.prototype["setBitmap"]=function(bitmap){
		this.bitmap = bitmap;
};
	
SFBitmapFunctionPerlinNoise.prototype["getOctaveValue"]=function(u,v,octave){
		var U=u*octave;
		U-=(Math.floor(U));
		var V=v*octave;
		V-=(Math.floor(V));
		
		var value=bitmap.getValue(U, V);
		return value*256;
};
	

